shuriken/include/shuriken/common/extract_file.h

```diff
@@ -0,0 +1,21 @@
+//--------------------------------------------------------------------*- C++ -*-
+// Shuriken-Analyzer: library for bytecode analysis.
+// @author Farenain <kunai.static.analysis@gmail.com>
+//
+// @file extract_file.h
+// @brief Utility to extract zip files such as APKs and IPAs
+// types
+
+#ifndef SHURIKENLIB_EXTRACT_FILE_H
+#define SHURIKENLIB_EXTRACT_FILE_H
+
+#include "zip.h"
+#include <string>
+#include <vector>
+#include <fstream>
+#include <iostream>
+
+/// Function to extract a file from a zip archive
+bool extract_file(zip_t *archive, const char *filename, const std::string &output_path);
```

Comment:
When you remove the external header, you will have that the compiler complains that `zip_t` does not exists, you can use a trick like the one done in logging:

https://github.com/Shuriken-Group/Shuriken-Analyzer/blob/main/shuriken/include/shuriken/common/logger.h#L16-L18

https://github.com/Shuriken-Group/Shuriken-Analyzer/blob/main/shuriken/include/shuriken/common/logger.h#L63

You declare the type in the header, so the compiler does not complain.  And we can use it in other objects. Probably in the future we will move the `common` folder directly to `lib` because it shouldn't be exposed to user, but for the moment, we need to avoid those dependencies.